contracts/transmuter/facets/Redeemer.sol

```diff
@@ -107,10 +109,19 @@ contract Redeemer is IRedeemer {
         address[] memory forfeitTokens
     ) internal returns (address[] memory tokens, uint256[] memory amounts) {
         TransmuterStorage storage ts = s.transmuterStorage();
+        // Reentrant protection
+        // On the first call, _notEntered will be true
+        if (ts.statusRentrant == ENTERED) revert ReentrantCall();
+        // Any calls to nonReentrant after this point will fail
+        ts.statusRentrant = ENTERED;
+
         if (ts.isRedemptionLive == 0) revert Paused();
         if (block.timestamp > deadline) revert TooLate();
+
         uint256[] memory subCollateralsTracker;
         (tokens, amounts, subCollateralsTracker) = _quoteRedemptionCurve(amount);
+        // Check that the provided slippage tokens length is identical to the redeem one
+        if (amounts.length != minAmountOuts.length) revert InvalidLengths();
```

Comment:
here we in fact have the space to cache amounts.length so we can maybe do it while doing the check

Reply:
Let's also change the `@dev` condition at the top of the redeem function that says:

```
/// @dev The `minAmountOuts` list must reflect or be longer than the amount of `tokens` returned
```

Reply:
second comment is unresolved